ansible_collections/arista/avd/roles/eos_designs/templates/base/hardware.j2

```diff
@@ -3,9 +3,11 @@
 {# Using tmp variable to be able to sort final output and handle duplicate assignments gracefully #}
 {%     set tmp_speed_groups = {} %}
 {%     if switch.platform is arista.avd.defined %}
-{%         for speed in platform_speed_groups[switch.platform] | arista.avd.natural_sort %}
-{%             for speed_group in platform_speed_groups[switch.platform][speed] | arista.avd.natural_sort %}
-{%                 do tmp_speed_groups.update({speed_group: speed}) %}
+{%         for speed in platform_speed_groups | arista.avd.convert_dicts('name', 'speeds') | arista.avd.natural_sort('name') | selectattr('name','arista.avd.defined',switch.platform) %}
```

Comment:
I think the logic is wrong here. As far as I remember, secondary_keys work on the dictionary values which is a list.

Here ** platform_speed_groups** is dictionary with key: **<platform>** and value is a dictionary **< speed >: [ < speed_group >, < speed_group > ]**

So the secondary can not be used here, may be in the next loop.